contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,550 @@
+	// to all the pools if raid type is not specified there
+	// Compulsory field if any raidGroup is not given Type
+	DefaultRaidGroupType string `json:"defaultRaidGroupType"`
+
+	// OverProvisioning to enable over provisioning
+	// Optional -- defaults to false
+	OverProvisioning bool `json:"overProvisioning"`
+	// Compression to enable compression
+	// Optional -- defaults to off
+	// Possible values : lz, off
+	Compression string `json:"compression"`
+	// Resources are the compute resources required by the cstor-pool
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources"`
+	// AuxResources are the compute resources required by the cstor-pool pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources"`
+	// Tolerations, if specified, the pool pod's tolerations.
+	Tolerations []corev1.Toleration `json:"tolerations"`
+
+	// PriorityClassName if specified applies to this pool pod
+	// If left empty, DefaultPriorityClassName is applied.
+	// (See CStorPoolClusterSpec.DefaultPriorityClassName)
+	// If both are empty, not priority class is applied.
+	PriorityClassName string `json:"priorityClassName"`
+}
+
+// RaidGroup contains the details of a raid group for the pool
+type RaidGroup struct {
+	// Type is the raid group type
+	// Supported values are : stripe, mirror, raidz and raidz2
+
+	// stripe -- stripe is a raid group which divides data into blocks and
+	// spreads the data blocks across multiple block devices.
+
+	// mirror -- mirror is a raid group which does redundancy
+	// across multiple block devices.
+
+	// raidz -- RAID-Z is a data/parity distribution scheme like RAID-5, but uses dynamic stripe width.
+	// radiz2 -- TODO
+	// Optional -- defaults to `defaultRaidGroupType` present in `PoolConfig`
+	Type string `json:"type"`
+
+	// LoadType can have following values :
+	// readCache -- The raid group is a read cache.
+	// writeCache -- The raid group is a write cache.
+	// spare -- The raid group is spare.
+	// "" -- Empty value means this will be used a regular vdev to form pool.
+	// What about default value ? Can this be changed after some point of time.
+	LoadType string `json:"load_type"`
+	// BlockDevices contains a list of block devices that
+	// constitute this raid group.
+
+	// DiskCount applies to raidz1 and raidz2 topology only.
+	// For raidz1 a valid DiskCount value is 2^n +1 where n>0
+	// For raidz2 a valid diskCount value is 2^n+2 where n>1
+	DiskCount int `json:"disk_count"`
+
+	BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
+}
+
+// CStorPoolClusterBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolClusterBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
```

Comment:
why we need capacity? to handle resize of underlying disk? does NDM updates BD?

Reply:
If it is not used, will remove it. Once we fill the need of this, it can be introduced.